libs/ledger-live-common/src/families/bitcoin/wallet-btc/wallet.ts

```diff
@@ -269,35 +283,35 @@ class BitcoinLikeWallet {
       ];
     });
 
-    const lastOutputIndex = txInfo.outputs.length - 1;
+    const lastOutputIndex = orderedOutputs.length - 1;
 
     log("hw", `createPaymentTransaction`, {
       inputs,
       associatedKeysets,
       outputScriptHex,
-      ...(params.lockTime && { lockTime: params.lockTime }),
-      ...(params.sigHashType && { sigHashType: params.sigHashType }),
-      ...(params.segwit && { segwit: params.segwit }),
+      lockTime: params.lockTime,
+      sigHashType: params.sigHashType,
+      segwit: params.segwit,
       initialTimestamp,
-      ...(params.expiryHeight && { expiryHeight: params.expiryHeight }),
-      ...(txInfo.outputs[lastOutputIndex]?.isChange && {
-        changePath: `${fromAccount.params.path}/${fromAccount.params.index}'/${txInfo.changeAddress.account}/${txInfo.changeAddress.index}`,
-      }),
+      expiryHeight: params.expiryHeight,
+      changePath: orderedOutputs[lastOutputIndex]?.isChange
```

Comment:
as you've changed the output order, the "txInfo.outputs[lastOutputIndex]?.isChange" is not necessary the "orderedOutputs[lastOutputIndex]?.isChange".
we should use the "txInfo.outputs[lastOutputIndex]?.isChange" as before